packages/modules-sdk/src/medusa-module.ts

```diff
@@ -67,22 +169,45 @@ export class MedusaModule {
       moduleExports
     )
 
-    await moduleLoader({
-      container,
-      moduleResolutions,
-      logger,
-    })
+    try {
+      await moduleLoader({
+        container,
+        moduleResolutions,
+        logger,
+      })
+    } catch (err) {
+      errorLoading(err)
+      throw err
+    }
 
     const services = {}
 
-    for (const resolution of Object.values(moduleResolutions)) {
+    for (const resolution of Object.values(
+      moduleResolutions
+    ) as ModuleResolution[]) {
       const keyName = resolution.definition.key
       const registrationName = resolution.definition.registrationName
 
       services[keyName] = container.resolve(registrationName)
+      services[keyName].__definition = resolution.definition
+
+      if (resolution.definition.isQueryable) {
```

Comment:
question: What is an example of a non-queryable module?

Reply:
Event bus and Cache. In future `Locking` as well

Reply:
Makes sense - thanks!